run.go

```diff
@@ -107,6 +107,10 @@ func addHostsToFile(hosts []string) error {
 }
 
 func addCommonOptsToSpec(commonOpts *CommonBuildOptions, g *generate.Generator) error {
```

Comment:
I wonder if changing from `*CommonBuildOptions` to `CommonBuildOptions` (removing the asterisk) can solve the problem too:
1) It seems to me that we set Linux resources using it but we don't change `commonOpts` during that process so why using the pointer?
2) This will solve the checking if `nil`.
3) Empty `commonOpts` will set all its properties to `0` or to empty strings (depends on the type). So we don't need to initialize it.